feat(oauth): #COR-1425 Support max lifetime on access tokens (#563)
GitGuardian/ggshield: changelog.d/20230615_104700_arnaud.lustrat_support_max_lifetime_tokens.md

> ### Added

Reviewer: ```suggestion
### Changed
```

From the user point-of-view, this is not a new feature.

> - ggshield auth login` now displays a warning message if the token expiration date has been adjusted to comply with the personal access token maximum lifetime setting of the user's workspace.

Reviewer 2: ```suggestion
- `ggshield auth login` now displays a warning message if the token expiration date has been adjusted to comply with the personal access token maximum lifetime setting of the user's workspace.
```

> - Supporting business workspaces personal access tokens max lifetime setting. A warning message is added to the login output if the token expiration date has been downsized to match the setting value.

Reviewer: ```suggestion
- `ggshield auth login` now displays a warning message if the token expiration date has been adjusted to comply with the personal access token maximum lifetime setting of the user's workspace.
```
